kafka-rest/src/main/java/io/confluent/kafkarest/DefaultKafkaRestContext.java

```diff
@@ -27,32 +33,35 @@
 public class DefaultKafkaRestContext implements KafkaRestContext {
 
   private final KafkaRestConfig config;
-  private ProducerPool producerPool;
   private KafkaConsumerManager kafkaConsumerManager;
+
   private Admin adminClient;
+  private Producer<byte[], byte[]> producer;
 
+  /**
+   * @deprecated Use {@link #DefaultKafkaRestContext(KafkaRestConfig)} instead.
+   */
+  @Deprecated
   public DefaultKafkaRestContext(
       KafkaRestConfig config,
       ProducerPool producerPool,
       KafkaConsumerManager kafkaConsumerManager
   ) {
-    this.config = config;
-    this.producerPool = producerPool;
-    this.kafkaConsumerManager = kafkaConsumerManager;
+    this(config);
   }
 
+  public DefaultKafkaRestContext(KafkaRestConfig config) {
+    this.config = requireNonNull(config);
+  }
 
   @Override
   public KafkaRestConfig getConfig() {
     return config;
   }
 
   @Override
-  public synchronized ProducerPool getProducerPool() {
-    if (producerPool == null) {
-      producerPool = new ProducerPool(config);
-    }
-    return producerPool;
+  public ProducerPool getProducerPool() {
```

Comment:
it seems like you can get rid of this too.

Reply:
Ditto confluent-security-plugins.